### PR TITLE
Preset for withdrawing multiple tokens to the same receipient

### PR DESCRIPTION
**Pull request by envin3**
Example with 2 tokens:
image
test done in https://dashboard.tenderly.co/explorer/vnet/2cceee6a-78e7-4d7d-a432-675e27b83082/transactions?kind=standard

### CODE REVIEW

File: src/settings/index.js [hunk 1]
```diff
@@ -87,6 +87,26 @@ const settings = {
       borrowingManagerClaimEnabled: true,
       feesManagerClaimEnabled: true,
       symbolPrice: 'USDC'
+    },
+    {
+      address: '0xdAC17F958D2ee523a2206206994597C13D831ec7',
```
Comment (gitmp01) [inline src/settings/index.js:92]: Checked :white_check_mark:

File: src/settings/index.js [hunk 2]
```diff
@@ -87,6 +87,26 @@ const settings = {
...
+      name: 'Tether USD',
+      decimals: 6,
+      symbol: 'USDT',
+      logo: './assets/svg/USDT.svg',
+      borrowingManagerClaimEnabled: true,
+      feesManagerClaimEnabled: true,
+      symbolPrice: 'USDT'
+    },
+    {
+      address: '0xB50721BCf8d664c30412Cfbc6cf7a15145234ad1',
```
Comment (gitmp01) [inline src/settings/index.js:102]: Checked :white_check_mark:

File: src/utils/presets/utils.js [hunk 3]
```diff
@@ -52,10 +58,10 @@ export const prepareWithdrawInflation = (ethPNTAddress, rawAmount) => [
   }
 ]
 
-export const prepareTransfer = (ethPNTAddress, receiverAddress, rawAmount) => [
+export const prepareTransfer = (tokenAddress, receiverAddress, rawAmount) => [
   {
     to: settings.contracts.financeVault,
-    calldata: vaultContract.encodeFunctionData('transfer', [ethPNTAddress, receiverAddress, rawAmount])
+    calldata: vaultContract.encodeFunctionData('transfer', [tokenAddress, receiverAddress, rawAmount])
```
Comment (gitmp01) [inline src/utils/presets/utils.js:73]: :+1:

File: src/utils/presets/utils.js [hunk 4]
```diff
@@ -66,6 +72,28 @@ export const prepareInflationProposal = async (ethPNTAddress, receiverAddress, r
...
+  defaultInput,
+  defaultInputLength,
+  inputList,
+  inputListLength
+) => {
+  if (!number || isNaN(number) || ethers.utils.isAddress(number)) return defaultInput(presetParams, setPresetParams)
+  else {
+    const indices = Array.from({ length: number }, (_, index) => defaultInputLength + index * inputListLength)
+    console.log('indices', indices, defaultInputLength, inputListLength)
+    console.log(typeof inputList)
```
Comment (gitmp01) [inline src/utils/presets/utils.js]: Wanted logging?

**Review by gitmp01 — APPROVED** [review]
LGTM

File: src/utils/presets/utils.js (hunk 4 above)
Comment (envin3) [reply src/utils/presets/utils.js]: removed, thanks

**Comment by envin3** [comment]
As possible to review [here](https://dashboard.tenderly.co/explorer/vnet/99d2aac5-9e51-4cdf-9d10-8b02102fc797/tx/0x97c79682b034c2e032b65932c1a2c38ab1371da31334f75ece06a6be53433198 ) it is not possible to send Native assets to a contract (like a gnosis safe).
It has then been added the possibility to send eth to a different address than erc20 tokens. In this way it is possible to send erc20 tokens also to contracts (lika a gnosis safe), while sending native assets (ETH) to a normal account.

tested in: https://dashboard.tenderly.co/explorer/vnet/5ae95343-1e02-4586-867d-a074785f5c6d/transactions?kind=standard

In case the destination address for Native assets is a contract a error is raised and the tx do not go through.

image


image

**Review by gitmp01 — APPROVED** [review]
LGTM

**Pull request merged**